### PR TITLE
fix: Use consistent error type between doc, logger and error class

### LINKED ISSUE #5045: Use consistent error type between doc, logger and error class
### What is the improvement or update you wish to see?

There is an inconsistent use of two of the error codes, `OAUTH_CALLBACK_ERROR` and `CALLBACK_OAUTH_ERROR`, of which only the latter is documented as errors: https://next-auth.js.org/errors.

In https://github.com/nextauthjs/next-auth/pull/5027#issuecomment-1197053995, @ThangHuuVu has suggested that we should:

1. merge docs of the two types mentioned above,
2. rename them more generic something like `CALLBACK_ERROR`, and
3. keep from the same error logged twice.

As a result we can merit from getting rid of subtly discrepant errors, both from docs page and from the source.

In the PR mentioned above, I had attempted to use only `CALLBACK_OAUTH_ERROR`, but after that I found there is the `OAuthCallbackError` class. Consequently I'm thinking to use `OAUTH_CALLBACK_ERROR` so that documentation better represents the implementation.

### Is there any context that might help us understand?

After receiving a comment in #5027 I've researched a bit and found some things that might affect the way we sort our docs.

So currently we have in the errors page a headline that says:

> oauth_callback_error expected 200 OK with body but no body was returned

while in fact it seems this `expected 200 OK with body but no body was returned` message never happens in callback phase, but in discovery phase. That is, openid-client emits this message when (and apparently only when) `/.well-known/openid-configuration` returns no body. cf. https://github.com/panva/node-openid-client/blob/v5.1.8/lib/helpers/process_response.js#L59

In next-auth's side, things described below in order happens

1. triggers issuer discovery
2. fails to get authorization url
4. logs `SIGNIN_OAUTH_ERROR` with`expected 200 OK with body but no body was returned`

So I believe the original content which attributes this error message to `OAUTH_CALLBACK_ERROR` was not true. I guess associating it with `SIGNIN_OAUTH_ERROR` can be of help and so I'm planning to create a PR.

### Does the docs page already exist? Please link to it.

https://next-auth.js.org/errors

### PR DESCRIPTION
**Pull request by sato11**
## ☕️ Reasoning
Following up this comment: https://github.com/nextauthjs/next-auth/pull/5027#issuecomment-1197053995.

I've abstained from further renaming it like `CALLBACK_ERROR`, since my original intention is only to make them consistent. I'm feeling happy enough with `OAUTH_CALLBACK_ERROR`, partly because I am not really sure over the intention of renaming. So I'd like to leave that to you.

About the doc, two changes have been made: merging `CALLBACK_OAUTH_ERROR` and `oauth_callback_error`, and relocating `openid-client` stuff to `SIGNIN_OAUTH_ERROR` section. Background of the latter change is reasoned in #5045. 

Commits are separately made, but if you want me to squash them let me know.

## 🧢 Checklist

- [x] Documentation
- [x] Tests
- [ ] Ready to be merged

## 🎫 Affected issues

Closes: #5045

## 📌 Resources

- [Contributing guidelines](./CONTRIBUTING.md)
- [Code of conduct](./CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

### CODE REVIEW

File: packages/next-auth/src/core/routes/callback.ts [hunk 1]
```diff
@@ -190,7 +190,7 @@ export default async function callback(params: {
       }
     } catch (error) {
       if ((error as Error).name === "OAuthCallbackError") {
-        logger.error("CALLBACK_OAUTH_ERROR", error as Error)
+        logger.error("OAUTH_CALLBACK_ERROR", error as Error)
```
Comment (ThangHuuVu) [inline packages/next-auth/src/core/routes/callback.ts:193]:
```suggestion
        logger.error("OAUTH_CALLBACK_ERROR", {
          error: error as Error,
          providerId: provider.id,
        })
```
Let's provide the `providerId` here 🙌

Comment (sato11) [reply packages/next-auth/src/core/routes/callback.ts:193]: Sure. Committing your suggestion 🙂

**Comment by sato11** [comment]
@ThangHuuVu Thanks for the feedback. How does it look to you now?

**Review by ThangHuuVu — APPROVED** [review]
LGTM 🚀

**Comment by ThangHuuVu** [comment]
@ndom91 @lluia I was wondering if changing the error code requires a bump in semver 🤔 WDYT?

**Pull request merged**